azuredevops/internal/service/permissions/resource_git_permissions.go

```diff
@@ -0,0 +1,210 @@
+package permissions
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/ahmetb/go-linq"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/microsoft/azure-devops-go-api/azuredevops/git"
+	"github.com/terraform-providers/terraform-provider-azuredevops/azuredevops/internal/client"
+	securityhelper "github.com/terraform-providers/terraform-provider-azuredevops/azuredevops/internal/service/permissions/utils"
+	"github.com/terraform-providers/terraform-provider-azuredevops/azuredevops/internal/utils/converter"
+	"github.com/terraform-providers/terraform-provider-azuredevops/azuredevops/internal/utils/debug"
+	"github.com/terraform-providers/terraform-provider-azuredevops/azuredevops/internal/utils/validate"
+)
+
+// ResourceGitPermissions schema and implementation for Git repository permission resource
+func ResourceGitPermissions() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceGitPermissionsCreate,
+		Read:   resourceGitPermissionsRead,
+		Update: resourceGitPermissionsUpdate,
+		Delete: resourceGitPermissionsDelete,
+		Importer: &schema.ResourceImporter{
+			State: resourceGitPermissionsImporter,
+		},
+		Schema: securityhelper.CreatePermissionResourceSchema(map[string]*schema.Schema{
+			"project_id": {
+				Type:         schema.TypeString,
+				ValidateFunc: validate.UUID,
+				Required:     true,
+				ForceNew:     true,
+			},
+			"repository_id": {
+				Type:         schema.TypeString,
+				ValidateFunc: validate.UUID,
+				Optional:     true,
+				ForceNew:     true,
+			},
+			"branch_name": {
+				Type:         schema.TypeString,
+				ValidateFunc: validate.NoEmptyStrings,
+				Optional:     true,
+				ForceNew:     true,
+				RequiredWith: []string{"repository_id"},
+			},
+		}),
+	}
+}
+
+func createGitToken(clients *client.AggregatedClient, d *schema.ResourceData) (*string, error) {
```

Comment:
Super nit-pick: Could you move the functions so that the functions called by the resource are first?	
```golang
                Create: resourceGitPermissionsCreate,
		Read:   resourceGitPermissionsRead,
		Update: resourceGitPermissionsUpdate,
		Delete: resourceGitPermissionsDelete,
```